genvids.py

```diff
@@ -0,0 +1,217 @@
+    else:  # Assume macOS
+        commands.append("open")
+
+    commands.append(path)
+
+    FNULL = open(os.devnull, 'w')
+    subprocess.call(commands, stdout=FNULL, stderr=subprocess.STDOUT)
+    FNULL.close()
+
+
+def manim(af, vidcache, hard, tc, manim_args, copy=False):
+    if not os.path.exists(vidcache):
+        os.makedirs(vidcache)
+
+    # extend python path to inclde this directory
+    env = os.environ.copy()
+    ppath = ""
+    if "PYTHONPATH" in env:
+        ppath = env["PYTHONPATH"]
+    env["PYTHONPATH"] = f"{af.dirname}:.:{ppath}"
+
+    args = ["manim", af.srcfile, ANIMATION_CLASS_NAME,
+            "--video_output_dir", af.video_output_dir,
+            "--media_dir", vidcache]
+    args.extend(manim_args)
+
+    gen = True
+    filepath = ""
+    manim_file_name = ""
+    if "--save_last_frame" in args:
+        gen = af.genimg
+        filepath = af.imgloc
+        manim_file_name = af.manim_file_name_img
+    else:
+        gen = af.genvid
+        filepath = af.vidloc
+        manim_file_name = af.manim_file_name_vid
+
+    args.extend(["--file_name", manim_file_name])
+
+    if hard or gen:
+        sys.stdout.write(f"{tc.BLUE}[RUNNING]{tc.ENDC} {af.srcfile} ~> {filepath}")
+        sys.stdout.flush()
+
+        r = subprocess.run(args, capture_output=True, env=env)
+        sys.stdout.write('\r')
+        sys.stdout.flush()
+
+        if r.returncode == 0:
+            print(f"{tc.GREEN}[SUCCESS]{tc.ENDC} {af.srcfile} ~> {filepath}")
+        else:
+            print(f"{tc.RED}[FAIL]{tc.ENDC}    {af.srcfile} ~> {filepath}")
+
+        print("STD OUT")
+        print(f"{r.stdout.decode()}")
+        print("STD ERR")
+        print(f"{r.stderr.decode()}")
+    else:
+        print(f"{tc.YELLOW}[CACHE]{tc.ENDC}   {af.srcfile} ~> {filepath}")
+        if "--preview" in args:
+            preview_file(filepath)
+    if copy:
+        dstfile = os.path.join("website", "public", af.public_path_vid)
+        dstdir, _ = os.path.split(dstfile)
+        if not os.path.exists(dstdir):
+            os.makedirs(dstdir)
+        shutil.copy(filepath, dstfile)
+        print(f"{tc.PURPLE}[COPY]{tc.ENDC}        ~> {dstfile}")
+
+
+def find_all(anim_dir, out_dir, hq):
+    anims = []
+    for dirname, subdir, filelist in os.walk(anim_dir):
+        for fname in filelist:
+            if fname.endswith("_anim.py"):
+                anims.append(AnimationFile(out_dir, dirname, fname, hq))
+    return anims
+
+
+if __name__ == "__main__":
+    DESCRIPTION = """
+        Generate animations for 'manim', will generate videos for any python
+        file with the suffix *_anim.py. Only the `Animation` class will be
+        compiled per individual filesh.
+
+        For any files in the animation tree, the output will be in the directory
+        specified by --out with the same directory structure, called *_anim.mp4,
+        in accordance with the python file.
+    """
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
```

Comment:
Nit (but will merge anyway without a fix): A few common conventions in Python scripts
* Have an explicit `main` method that this `if` statement calls
* Have a method to build the `ArgumentParser` to get those low-level details out of main.